### PR TITLE
Unit test for 3P app 

### PR DESCRIPTION
**Pull request by khushi2305**
This PR contains the unit test for functions used by the app -
1) Test handling responses of SDK functions 
2) Test for verifying that the offer is valid or not 
3) Test for checking that the username displayed is correct or not

### CODE REVIEW

**Comment by khushi2305** [comment]
@googlebot I consent.

File: home_insurance_app/lib/data/database_utilities.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
```
Comment (dreamplay) [inline home_insurance_app/lib/data/database_utilities.dart:1]: database_utilities is not n appropriate name

File: home_insurance_app/lib/data/helper_functions.dart [hunk 2]
```diff
@@ -5,23 +5,41 @@ import 'package:optional/optional.dart';
 import 'package:flutter/material.dart';
 import 'package:homeinsuranceapp/pages/list_structures.dart';
```
Comment (dreamplay) [inline home_insurance_app/lib/data/helper_functions.dart:6]: helper_functions isnt an appropriate name

**Comment by khushi2305** [comment]
@googlebot I consent .

File: home_insurance_app/lib/data/helper_functions.dart (hunk 2 above)
Comment (khushi2305) [reply home_insurance_app/lib/data/helper_functions.dart]: This file is already renamed . Removed the import

**Comment by anjalisoni3655** [comment]
@googlebot I consent.

File: home_insurance_app/lib/data/offer_service.dart [hunk 3]
```diff
@@ -31,7 +31,7 @@ Future<void> getDevices() async {
   }
 }
 
-// If has access but deosnt have list of structures retries to fetch list of structures
+// If has access but doesnt have list of structures retries to fetch list of structures
```
Comment (akashagarwal) [inline home_insurance_app/lib/data/offer_service.dart:34]: nit: it [] has access to [] but...

**Comment by khushi2305** [comment]
@googlebot I consent.

**Pull request merged**